pkg/supportbundle/supportbundle_test.go

```diff
@@ -7,40 +7,71 @@ import (
-    if err != nil {
-        t.Error("couldn't parse bundle 2")
-    }
-
-    fulldoc,err := LoadSupportBundleSpec("test/completebundle.yaml")
-    if err != nil {
-        t.Error("couldn't load full bundle from file")
-    }
-
-    fullbundle,err := ParseSupportBundleFromDoc(fulldoc)
-    if err != nil {
-        t.Error("couldn't parse full bundle")
-    }
-
-    bundle3 := ConcatSpec(bundle1,bundle2)
-
-    if reflect.DeepEqual(fullbundle, bundle3) == false {
-        t.Error("Full bundle and concatenated bundle are not the same.")
-    }
+	bundle1doc, err := LoadSupportBundleSpec("test/supportbundle1.yaml")
+	if err != nil {
+		t.Error("couldn't load bundle1 from file")
+	}
+
+	bundle2doc, err := LoadSupportBundleSpec("test/supportbundle2.yaml")
+	if err != nil {
+		t.Error("couldn't load bundle2 from file")
+	}
+
+	bundle1, err := ParseSupportBundleFromDoc(bundle1doc)
+	if err != nil {
+		t.Error("couldn't parse bundle 1")
+	}
+
+	bundle2, err := ParseSupportBundleFromDoc(bundle2doc)
+	if err != nil {
+		t.Error("couldn't parse bundle 2")
+	}
+
+	fulldoc, err := LoadSupportBundleSpec("test/completebundle.yaml")
+	if err != nil {
+		t.Error("couldn't load full bundle from file")
+	}
+
+	fullbundle, err := ParseSupportBundleFromDoc(fulldoc)
+	if err != nil {
+		t.Error("couldn't parse full bundle")
+	}
+
+	bundle3 := ConcatSpec(bundle1, bundle2)
+
+	if reflect.DeepEqual(fullbundle, bundle3) == false {
+		t.Error("Full bundle and concatenated bundle are not the same.")
+	}
+
+	//a spec with a uri should return the upstream spec and not the one in the given doc
+
+	providedBundle, _ := LoadSupportBundleSpec("test/uri-spec.yaml")
+	providedSpec, _ := ParseSupportBundle(providedBundle, true)
+	upstreamBundle, _ := LoadSupportBundleSpec("https://raw.githubusercontent.com/replicatedhq/troubleshoot-specs/main/host/cluster-down.yaml")
+	upstreamSpec, _ := ParseSupportBundle(upstreamBundle, false)
+
+	if reflect.DeepEqual(providedSpec, upstreamSpec) == false {
+		t.Errorf("expected provided bundle to be replaced with upstream bundle. expected %v but got %v", upstreamSpec, providedSpec)
+	}
+
+	//if we can't reach the uri, we should fall back to the provided spec
+	badURIBundle, _ := LoadSupportBundleSpec("test/bad-uri-spec.yaml")
+	badURISpec, _ := ParseSupportBundleFromDoc(badURIBundle)
+	providedBundle, _ = LoadSupportBundleSpec("test/bad-uri-spec.yaml")
+	providedSpec, _ = ParseSupportBundle(providedBundle, false)
+
+	if reflect.DeepEqual(providedSpec, badURISpec) == false {
+		t.Errorf("expected bad URI to fall back to provided bundle.  expected %v but got %v", providedSpec, badURISpec)
+	}
+
+	//if we can't parse the upstream doc, we should fall back to the provided
+	badYAMLBundle, _ := LoadSupportBundleSpec("test/bad-upstream-uri-spec.yaml")
+	badYAMLSpec, _ := ParseSupportBundleFromDoc(badYAMLBundle)
+	providedBundle, _ = LoadSupportBundleSpec("test/bad-upstream-uri-spec.yaml")
+	providedSpec, _ = ParseSupportBundle(providedBundle, false)
```

Comment:
Wonder if this should be in a separate test since we aren't testing concatenating here?

Reply:
yeah makes sense, I need to move the test specs out of my fork, too